go-controller/pkg/node/base_node_network_controller_dpu.go

```diff
@@ -14,139 +13,214 @@ import (
+	klog.Infof("Adding %s on DPU", podDesc)
+	podInterfaceInfo, err := cni.PodAnnotation2PodInfo(pod.Annotations, nil, isOvnUpEnabled,
+		string(pod.UID), "", nadName, netName, config.Default.MTU)
+	if err != nil {
+		klog.Errorf("Failed to get pod interface information of %s: %v. retrying", podDesc, err)
+		return err
+	}
+	err = bnnc.addRepPort(pod, dpuCD, podInterfaceInfo, getter)
+	if err != nil {
+		klog.Errorf("Failed to add rep port for %s, %v. retrying", podDesc, err)
+	}
+	return err
+}
+
+func dpuConnectionDetailChanged(oldDPUCD, newDPUCD *util.DPUConnectionDetails) bool {
+	if oldDPUCD == nil && newDPUCD == nil {
+		return false
+	}
+	if (oldDPUCD != nil && newDPUCD == nil) || (oldDPUCD == nil && newDPUCD != nil) {
+		return true
+	}
+	if oldDPUCD.PfId != newDPUCD.PfId ||
+		oldDPUCD.VfId != newDPUCD.VfId || oldDPUCD.SandboxId != newDPUCD.SandboxId {
+		return true
+	}
+	return false
+}
+
 // watchPodsDPU watch updates for pod dpu annotations
 func (bnnc *BaseNodeNetworkController) watchPodsDPU() error {
-	var retryPods sync.Map
-	// servedPods tracks the pods that got a VF
-	var servedPods sync.Map
-
 	clientSet := cni.NewClientSet(bnnc.client, corev1listers.NewPodLister(bnnc.watchFactory.LocalPodInformer().GetIndexer()))
 
-	// Support default network for now
-	nadName := types.DefaultNetworkName
 	netName := types.DefaultNetworkName
 	_, err := bnnc.watchFactory.AddPodHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			pod := obj.(*kapi.Pod)
-			klog.Infof("Add for Pod: %s/%s", pod.ObjectMeta.GetNamespace(), pod.ObjectMeta.GetName())
+			klog.V(5).Infof("Add for Pod: %s/%s for network %s", pod.Namespace, pod.Name, netName)
 			if util.PodWantsHostNetwork(pod) || pod.Status.Phase == kapi.PodRunning {
 				return
 			}
-			if util.PodScheduled(pod) {
-				// Is this pod created on same node as the DPU
-				if bnnc.name != pod.Spec.NodeName {
-					return
-				}
+			// add all the Pod's NAD into Pod's podNadCache
+			// networkMap is network selection element map, key is the NAD name. For default network,
+			// NAD name is DefaultNetworkName.
+			networkMap := make(map[string]*nettypes.NetworkSelectionElement)
+			// Support default network for now
+			networkMap[types.DefaultNetworkName] = nil
+			bnnc.podNadCache.Store(pod.UID, networkMap)
 
-				vfRepName, err := bnnc.getVfRepName(pod)
-				if err != nil {
-					klog.Infof("Failed to get rep name, %s. retrying", err)
-					retryPods.Store(pod.UID, true)
-					return
-				}
-				isOvnUpEnabled := atomic.LoadInt32(&bnnc.atomicOvnUpEnabled) > 0
-				podInterfaceInfo, err := cni.PodAnnotation2PodInfo(pod.Annotations, nil, isOvnUpEnabled, string(pod.UID),
-					"", nadName, netName, config.Default.MTU)
-				if err != nil {
-					retryPods.Store(pod.UID, true)
-					return
+			// initialize serverCache to be empty
+			isOvnUpEnabled := atomic.LoadInt32(&bnnc.atomicOvnUpEnabled) > 0
+			servedCache := map[string]*util.DPUConnectionDetails{}
```

Comment:
`servedCache` is a confusing name. It is a map of nad => DPU Connection Details. Why not some name that describes what it is E.g. `NadToDpuCDMap`